questions/services.py

```diff
@@ -663,6 +672,73 @@ def create_forecast_bulk(*, user: User = None, forecasts: list[dict] = None):
             raise ValidationError("You cannot forecast on this question yet!")
 
         create_forecast(question=question, user=user, **forecast)
+        after_forecast_actions(question, user)
+
+    # Running forecast post triggers
+    for post in posts:
+        # There may be situations where async jobs from `create_forecast` complete after
+        # `run_on_post_forecast` is triggered. To maintain the correct sequence of execution,
+        # we need to ensure that `run_on_post_forecast` runs only after all forecasts have been processed.
+        #
+        # As a temporary solution, we introduce a 10-second delay before execution
+        # to ensure all forecasts are processed.
+        run_on_post_forecast.send_with_options(args=(post.id,), delay=10_000)
+
+
+def withdraw_forecast_bulk(user: User = None, withdrawals: list[dict] = None):
+    from posts.services.common import get_post_permission_for_user
+
+    posts = set()
+
+    for withdrawal in withdrawals:
+        question = cast(Question, withdrawal["question"])
+        post = question.get_post()
+        posts.add(post)
+
+        # Check permissions
+        permission = get_post_permission_for_user(post, user=user)
+        ObjectPermission.can_forecast(permission, raise_exception=True)
```

Comment:
I see we already performed such checks in `bulk_withdraw_forecasts_api_view `

Reply:
This is true.
The duplication is in `bulk_create_forecasts_api_view`, which I was copying.
I'll remove duplication from both.